src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2520,12 +2520,13 @@ namespace ListIndex {
 static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
     ASRUtils::require_impl(x.n_args <= 4, "Call to list.index must have at most four arguments",
         x.base.base.loc, diagnostics);
-    ASRUtils::require_impl(ASR::is_a<ASR::List_t>(*ASRUtils::expr_type(x.m_args[0])) &&
-        ASRUtils::check_equal_type(ASRUtils::expr_type(x.m_args[1]),
-            ASRUtils::get_contained_type(ASRUtils::expr_type(x.m_args[0]))),
-        "First argument to list.index must be of list type and "
-        "second argument must be of same type as list elemental type",
-        x.base.base.loc, diagnostics);
+    ASR::ttype_t *list_type = ASRUtils::type_get_past_const(ASRUtils::expr_type(x.m_args[0]));
+    ASRUtils::require_impl(ASR::is_a<ASR::List_t>(*list_type) &&
+                               ASRUtils::check_equal_type(ASRUtils::expr_type(x.m_args[1]),
+                                                          ASRUtils::get_contained_type(list_type)),
+                           "First argument to list.index must be of list type and "
+                           "second argument must be of same type as list elemental type",
+                           x.base.base.loc, diagnostics);
```

Comment:
Why formatting change here? It makes it harder to understand what was changed.

Reply:
I am sorry, the IDE did it on it's own.

Reply:
Done.